zipline/utils/api_utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from six import iteritems
+from toolz import valmap
+
+from zipline.utils.preprocess import preprocess
+
+
+def ensure_upper_case(func, argname, arg):
+    if isinstance(arg, str):
```

Comment:
Right now it will fail for unicode args, which seems undesirable. What is the best way to additionally accept unicode? Something like the following:

```
if isinstance(arg, str) or isinstance(arg, unicode):
    return arg.upper()
```